8.x/theme-development/representative.md

```diff
@@ -0,0 +1,381 @@
+
+![Conditional option1](/repcondition.png)
+
+Another example is to build a conditional option to show a specific field when a specific options is selected.
+For example if you set current customers as representative, you will need a field for that customer information(for example, the phone number). So the number input will be displayed only when the current customer is selected. Otherwise this input won't be shown. See its step-by-step instruction:
+
+
+- First, create an option and give it a name attribute:
+
+```php{7}
+@if(representative_is_enabled())
+    <div class="col-md-6 col-md-6 col-xs-12">
+        <div class="form-group">
+            <label for="representative_type">نحوه آشنایی با مجموعه</label>
+            <select class="form-control number-control" name="representative_type">
+                <option value="">هیج کدام</option>
+                <option value="مشتریان فعلی مجموعه" id="manual-representative">مشتریان فعلی مجموعه</option>
+                @foreach(representative_get_options() as $option)
+                    <option value="{{$option}}">{{$option}}</option>
+                @endforeach
+            </select>
+        </div>
+    </div>
+    </div>
+```
+**Note:** this is a manual option and should be written out of the foreach loop.
+
+
+- Then add a condition to the options :
+
+```php
+ @if(representative_is_customer_representative_enabled())
+     <div class="col-md-6 col-sm-6 col-xs-12" id="representative-input-container" style="display: none">
+        <div class="form-group">
+            <label for="family">شماره تماس معرف در صورت تمایل</label>
+                <input type="text" class="form-control number-control" name=representative-username"
+                placeholder="شماره تماس معرف" 
+                value="{{old('representative-username')}} " > 
+        </div>        
+     </div>
+ @endif
+```
+The very next step is to add `javaScript` codes to set the apearance and adjust the options elements:
+
+```php
+ //larammerce-theme/resources/js/require/register-auth-require
+
+if (window.currentPage === "auth-register")
+    require(["jquery"], function (jQuery) {
+        const representativeSelectEl = jQuery("select[name='representative_type']");
+        const representativeInputContainerEl = jQuery("#representative-input-container");
+        if (representativeSelectEl.length > 0 && representativeInputContainerEl.length > 0) {
+            representativeSelectEl.on("change", function (event) {
+                const optionSelected = jQuery("option:selected", this);
+                const id = optionSelected.attr("id");
+
+                if (id === "manual-representative") {
+                    representativeInputContainerEl.fadeIn();
+                }else{
+                    representativeInputContainerEl.fadeOut();
+                }
+            });
+        }
+    });
+```
```

Comment:
Now deploy the larammerce-theme project with `./deploy.sh` and refresh the registration page to see the result.